docs/spfx/toolchain/implement-ci-cd-with-azure-devops.md

```diff
@@ -0,0 +1,207 @@
+---
+title: Implement Continuous Integration and Continuous deployment using Azure DevOps
+description: Streamlining the build and deployment process by automating manual steps.
+ms.date: 09/06/2018
+ms.prod: sharepoint
+---
+
+
+# Implement Continuous Integration and Continuous deployment using Azure DevOps
+Azure DevOps (Visual Studio Team Services / Team Foundation Server) consists of a set of tools and services that help developers implement DevOps, Continuous Integration, and Continuous Deployment processes for their development projects.
+
+This article explains the steps involved in setting up your Azure DevOps environment with with Continuous Integration and Continuous Deployment to automate your SharePoint Framework builds, unit tests, and deployment.
+
+## Continuous Integration
+Continuous Integration (CI) helps developers integrate code into a shared repository by automatically verifying the build using unit tests and packaging the solution each time new code changes are submitted.  
+
+Setting up Azure DevOps for Continuous Integration with a SharePoint Framework solution requires the following steps:
+1. Creating the Build Definition
+2. Installing NodeJS
+3. Restoring dependencies
+4. Executing Unit Tests
+5. Importing code coverage information
+6. Bundling the solution
+7. Packaging the solution
+8. Preparing the artifacts
+9. Publishing the artifacts
+
+### Creating the Build Definition
+The Build Definition, as its name suggests, includes all the definitions and their configurations for the build.  Start setting up your Continuous Integration by creating a new build definition and link it to your repository.
+![linking the build definition to the repository](../../images/azure-devops-spfx-01.png)
+> [!NOTE] 
+> Build definitions can be described as a process template. It is a set of configured tasks that will be executed one after another on the source code every time a build is triggered. Tasks can be grouped in phases, by default a build definition contains at least one phase. You can add new tasks to the phase by clicking on the big plus sign next to the phase name.
+
+### Installing NodeJS version 8
```

Comment:
In the referenced pull request SharePoint/sp-dev-docs/#2549 with the exported json build pipeline you reference already to NodeJS version 10. Wouldn't that be better, to also reflect that in the documentation?

Reply:
so we had a discussion with Waldek and the supported version at the moment is version 8. And I forgot to update the sample. Do you want to issue a PR on the sample or should I make the changes myself?

Reply:
I can make the PR

Reply:
see https://github.com/SharePoint/sp-dev-build-extensions/pull/27